lib/plug/router/utils.ex

```diff
@@ -67,6 +67,21 @@ defmodule Plug.Router.Utils do
   end
 
   @doc """
+  Builds a list of path param names and var match pairs that can bind to
+  dynamic path segment values. Excludes path segments with underscores.
+
+  ## Examples
+
+      iex> Plug.Router.Utils.build_path_params_match([:id])
+      [{"id", {:id, [], nil}}]
+  """
+  def build_path_params_match(vars) do
+    vars
+    |> Enum.map(fn(var) -> {Atom.to_string(var), Macro.var(var, nil)} end)
+    |> Enum.filter(fn({var, _macro}) -> String.at(var, 0) !== "_" end)
```

Comment:
I think your suggestion is readable enough, so I made the change. I also added to the `@doc` the reason **why** underscored path params have to be filtered out (compiler warning otherwise if you try to unquote them in the macro). Does that make sense?